src/test/mir-opt/funky_arms/rustc.float_to_exponential_common.ConstProp.diff

```diff
@@ -0,0 +1,158 @@
+  
+      bb6: {
+          StorageLive(_17);                // scope 2 at $DIR/funky_arms.rs:28:46: 28:49
+          _17 = &mut (*_1);                // scope 2 at $DIR/funky_arms.rs:28:46: 28:49
+          StorageLive(_18);                // scope 2 at $DIR/funky_arms.rs:28:51: 28:54
+          _18 = _2;                        // scope 2 at $DIR/funky_arms.rs:28:51: 28:54
+          StorageLive(_19);                // scope 2 at $DIR/funky_arms.rs:28:56: 28:60
+          _19 = _6;                        // scope 2 at $DIR/funky_arms.rs:28:56: 28:60
+          StorageLive(_20);                // scope 2 at $DIR/funky_arms.rs:28:62: 28:67
+          _20 = _3;                        // scope 2 at $DIR/funky_arms.rs:28:62: 28:67
+          _0 = const float_to_exponential_common_shortest::<T>(move _17, move _18, move _19, move _20) -> bb9; // scope 2 at $DIR/funky_arms.rs:28:9: 28:68
+                                           // ty::Const
+                                           // + ty: for<'r, 's, 't0> fn(&'r mut std::fmt::Formatter<'s>, &'t0 T, core::num::flt2dec::Sign, bool) -> std::result::Result<(), std::fmt::Error> {float_to_exponential_common_shortest::<T>}
+                                           // + val: Value(Scalar(<ZST>))
+                                           // mir::Constant
+                                           // + span: $DIR/funky_arms.rs:28:9: 28:45
+                                           // + literal: Const { ty: for<'r, 's, 't0> fn(&'r mut std::fmt::Formatter<'s>, &'t0 T, core::num::flt2dec::Sign, bool) -> std::result::Result<(), std::fmt::Error> {float_to_exponential_common_shortest::<T>}, val: Value(Scalar(<ZST>)) }
+      }
+  
+      bb7: {
+          StorageLive(_10);                // scope 2 at $DIR/funky_arms.rs:24:17: 24:26
+          _10 = ((_7 as Some).0: usize);   // scope 2 at $DIR/funky_arms.rs:24:17: 24:26
+          StorageLive(_11);                // scope 3 at $DIR/funky_arms.rs:26:43: 26:46
+          _11 = &mut (*_1);                // scope 3 at $DIR/funky_arms.rs:26:43: 26:46
+          StorageLive(_12);                // scope 3 at $DIR/funky_arms.rs:26:48: 26:51
+          _12 = _2;                        // scope 3 at $DIR/funky_arms.rs:26:48: 26:51
+          StorageLive(_13);                // scope 3 at $DIR/funky_arms.rs:26:53: 26:57
+          _13 = _6;                        // scope 3 at $DIR/funky_arms.rs:26:53: 26:57
+          StorageLive(_14);                // scope 3 at $DIR/funky_arms.rs:26:59: 26:72
+          StorageLive(_15);                // scope 3 at $DIR/funky_arms.rs:26:59: 26:68
+          _15 = _10;                       // scope 3 at $DIR/funky_arms.rs:26:59: 26:68
+          _14 = Add(move _15, const 1_usize); // scope 3 at $DIR/funky_arms.rs:26:59: 26:72
+                                           // ty::Const
+                                           // + ty: usize
+                                           // + val: Value(Scalar(0x00000001))
```

Comment:
Looks like this test needs the `// EMIT_MIR_FOR_EACH_BIT_WIDTH` directive.

Reply:
I changed the test to not have `usize` constants anymore instead.